plugins/source/azure/resources/services/subscription/subscription_mock_test.go

```diff
@@ -20,6 +21,8 @@ func createSubscriptions(router *mux.Router) error {
 
 	emptyStr := ""
 	item.NextLink = &emptyStr
+	item.Value[0].ID = to.Ptr("/subscriptions/sub-id")
```

Comment:
The reason the tests didn't catch this is because we mock all strings to the same value of `test string` so it doesn't capture the different format between ID and SubscriptionID